src/OmniSharp.Roslyn.CSharp/Services/Refactoring/V2/BaseCodeActionService.cs

```diff
@@ -25,23 +25,26 @@ public abstract class BaseCodeActionService<TRequest, TResponse> : IRequestHandl
         protected readonly OmniSharpWorkspace Workspace;
         protected readonly IEnumerable<ICodeActionProvider> Providers;
         protected readonly ILogger Logger;
-
-        private readonly CodeActionHelper _helper;
+        private readonly CSharpDiagnosticService analyzers;
+        private readonly CodeFixesForProjects codeFixesForProject;
         private readonly MethodInfo _getNestedCodeActions;
 
-        private static readonly Func<TextSpan, List<Diagnostic>> s_createDiagnosticList = _ => new List<Diagnostic>();
-
-        protected Lazy<List<CodeFixProvider>> OrderedCodeFixProviders;
         protected Lazy<List<CodeRefactoringProvider>> OrderedCodeRefactoringProviders;
 
-        protected BaseCodeActionService(OmniSharpWorkspace workspace, CodeActionHelper helper, IEnumerable<ICodeActionProvider> providers, ILogger logger)
+        // For some (probably visual studio specific?) reason diagnostic and fix codes doesn't match in every case.
+        private readonly Dictionary<string, string> customDiagVsFixMap = new Dictionary<string, string>
```

Comment:
we don't directly use CS8019 but has an analyzer that report different diagnsitic based on CS8019 to improve user experience.